gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -824,39 +824,34 @@ static LongrunningOperation parseLro(
     boolean isResponseTypeNameShortOnly = lastDotIndex < 0;
     String responseTypeShortName =
         lastDotIndex >= 0 ? responseTypeName.substring(lastDotIndex + 1) : responseTypeName;
+    // When only shortname is provided, match on same proto package as method (See
+    // https://aip.dev/151)
+    String responseTypeFullName =
+        isResponseTypeNameShortOnly
+            ? methodDescriptor.getFile().getPackage() + "." + responseTypeShortName
+            : responseTypeName;
 
     lastDotIndex = metadataTypeName.lastIndexOf('.');
     boolean isMetadataTypeNameShortOnly = lastDotIndex < 0;
     String metadataTypeShortName =
         lastDotIndex >= 0 ? metadataTypeName.substring(lastDotIndex + 1) : metadataTypeName;
+    // When only shortname is provided, match on same proto package as method (See
+    // https://aip.dev/151)
+    String metadataTypeFullName =
```

Comment:
What if the short name is invalid? Can we throw an exception in the generator? Or we are going to generate code that is not compilable?

Reply:
Note from discussion offline: if the short name is invalid (no match) then the exception is handled by existing logic here:
https://github.com/googleapis/sdk-platform-java/blob/b73d72aad94f0250841c1c8fcb0c2d3144237eea/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java#L858-L868

But this behavior should be tested (through true unit tests rather than goldens). Will look into replacing showcase-extended setup in this PR with true unit testing for `Parser`.